Add catalyst and adios

julienfausty opened the pull request:
This pull request comes in support of this one in GEOSX: https://github.com/GEOSX/GEOSX/pull/2250

It performs the following operations
Add new dependencies into `thirdPartyLibs` repository:
* `catalyst` itself version [2.0.0-rc3](https://gitlab.kitware.com/paraview/catalyst/-/tree/v2.0.0-rc3)
* `adios/catalyst` implementation on commit [af3b35f](https://gitlab.kitware.com/paraview/adioscatalyst/-/commit/af3b35fce79f7f947c7521c40106646204cbcd99)
* `adios2` [v2.8.3](https://github.com/ornladios/ADIOS2/tree/v2.8.3)
* update `conduit` from `0.8.2` to [0.8.6](https://github.com/LLNL/conduit/tree/v0.8.6)
* update `VTK` to [49361a2](https://gitlab.kitware.com/vtk/vtk/-/commit/49361a2058784db4be05980476457c06d86890e7) for mangling capabilities

Commit:
Catalyst: add catalyst and adios

Commit:
conduit: update version 0.8.2 -> 0.8.6

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -51,6 +51,7 @@ option( ENABLE_SCOTCH "" ON )
 option( ENABLE_SUITESPARSE "" ON )
 option( ENABLE_VTK "" ON )
 option( ENABLE_VTK_WRAP_PYTHON "" OFF )
+option( ENABLE_CATALYST "EEnable building catalyst2" ON)
```
typo EEnable

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -214,7 +215,7 @@ list(APPEND build_list hdf5 )
 ################################
 set(CONDUIT_DIR "${CMAKE_INSTALL_PREFIX}/conduit")
-set(CONDUIT_URL "${TPL_MIRROR_DIR}/conduit-0.8.2.tar.gz")
+set(CONDUIT_URL "${TPL_MIRROR_DIR}/conduit-0.8.6.zip")
```
this is not the Catalyst version, is this wanted ?

julienfausty replied:
Yes, I don't think we should tether geosx's `conduit`version to `catalyst`'s. This is just an update of the `conduit` package to get some things compiling

Commit:
VTK: update to master and add mangling

Mangling VTK uses an inline namespace to change the VTK symbols in the
binary so as to allow multiple VTK binaries to coexist

Commit:
GCC13: patch and fix errors and warnings

* downgrade fmt 10.0.0 -> 9.1.0
* patch raja for stdexcept and cstdlib

Commit:
ADIOSCatalyst: update version for M x N and multimesh fixes

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -1078,7 +1085,8 @@ endif()
 if( ENABLE_VTK )
     set( VTK_DIR "${CMAKE_INSTALL_PREFIX}/vtk" )
-    set( VTK_URL "${TPL_MIRROR_DIR}/VTK-9.2.6.tar.gz" )
+    set( VTK_URL "${TPL_MIRROR_DIR}/VTK-59d89108f4.tar.gz" )
```
you do not need to change, but a think a nightly (with a date as a version) is slightly more readable.
Can you comment that this is between 9.2.6 and 9.3.0 ?

julienfausty replied:
yep, I was not aware of the nightly VTK downloads, I will use that and update the repo

julienfausty replied:
see below

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -1088,6 +1096,12 @@ if( ENABLE_VTK )
+    # This mangling is needed in order to ensure that the VTK included with geosx does
+    # not interfere with another VTK version that may be linked dynamically at runtime
+    # (when using ParaView's Catalyst implementation for example).
+    set( VTK_ABI_NAMESPACE_NAME "vtk_geosx" )
```
do we need to "vtk_geosx" -> "vtk_geos" ?

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -1134,16 +1148,107 @@ endif( ENABLE_MPI )
 ################################
 set(FMT_DIR "${CMAKE_INSTALL_PREFIX}/fmt")
-set(FMT_URL "${TPL_MIRROR_DIR}/fmt-10.0.0.tar.gz")
+set(FMT_URL "${TPL_MIRROR_DIR}/fmt-9.1.0.zip")
```
why a downgrade ?

CharlesGueunet commented on CMakeLists.txt:
```diff
@@ -1106,9 +1120,9 @@ endif( ENABLE_MPI )
-                         PATCH_COMMAND patch -p0 < ${TPL_MIRROR_DIR}/vtkXMLReader-fpe.patch
                          BUILD_COMMAND ${TPL_BUILD_COMMAND}
                          INSTALL_COMMAND "${TPL_INSTALL_COMMAND}"
+                         PATCH_COMMAND patch -p1 < ${TPL_MIRROR_DIR}/VTK-ABI.patch
```
why is this a patch ? seems like it should be updated upstream

julienfausty replied:
This has been updated upstream, but to avoid wasting time with the merge I made this patch and then forgot about it. I will update VTK with a nightly build and take out the patch

julienfausty replied:
the `9.3.0-rc1` version has a broken ABI, let's wait for the release before making the changes

CharlesGueunet reviewed (approved):
LGTM after small questions / remarks

Commit:
ABI: quick naming geosx -> geos fix

CharlesGueunet committed:
Update adioscatalyst to multiple-writer branch

CharlesGueunet committed:
Merge remote-tracking branch 'public/master' into add-catalyst-and-adios

CharlesGueunet committed:
AdiosCatalyst master with multiple writer

CharlesGueunet committed:
Remote unnedded raja patches

CharlesGueunet committed:
Keep fmt version

CharlesGueunet committed:
Remove useless patches

CharlesGueunet committed:
Update adios catalyst to avid mpi issue

sframba committed:
Merge branch 'master' into add-catalyst-and-adios

sframba committed:
restored filename of fmt library archive

sframba committed:
Merge branch 'master' into add-catalyst-and-adios

sframba committed:
Merge branch 'master' into add-catalyst-and-adios
